### PR TITLE
Add an option to specify FFmpeg encoding parameters

### PR DESCRIPTION
**Pull request by varsill**

### CODE REVIEW

File: c_src/membrane_h264_ffmpeg_plugin/encoder.c [hunk 1]
```diff
@@ -123,6 +123,11 @@ UNIFEX_TERM create(UnifexEnv *env, int width, int height, char *pix_fmt,
 
   set_x264_defaults(&params, preset);
 
+  for(unsigned int i=0; i<ffmpeg_options_length; i++)
+  {
+    av_dict_set(&params, ffmpeg_options[i].key, ffmpeg_options[i].value, 0);
+  }
```
Comment (mat-hek) [inline c_src/membrane_h264_ffmpeg_plugin/encoder.c:129]: make sure that these options override the other options

File: lib/membrane_h264_ffmpeg/encoder.ex [hunk 2]
```diff
@@ -121,8 +121,21 @@ defmodule Membrane.H264.FFmpeg.Encoder do
                 See [this page](https://en.wikibooks.org/wiki/MeGUI/x264_Settings#scenecut) for more info.
                 """,
                 default: @default_sc_threshold
+              ],
+              ffmpeg_options: [
+                spec: %{String.t() => String.t()},
+                description: """
+                A map with options that will be passed to the encoder.
```
Comment (mat-hek) [inline lib/membrane_h264_ffmpeg/encoder.ex:128]: Let's mention that one shouldn't provide options that are in the top level here and add a check for that and a warn

**Review by mat-hek — CHANGES_REQUESTED** [review]

File: lib/membrane_h264_ffmpeg/encoder.ex (hunk 2 above)
Comment (varsill) [reply lib/membrane_h264_ffmpeg/encoder.ex]: done

File: c_src/membrane_h264_ffmpeg_plugin/encoder.c (hunk 1 above)
Comment (varsill) [reply c_src/membrane_h264_ffmpeg_plugin/encoder.c]: It indeed does ;)

File: lib/membrane_h264_ffmpeg/encoder.ex [hunk 3]
```diff
@@ -122,22 +123,29 @@ defmodule Membrane.H264.FFmpeg.Encoder do
...
+              ffmpeg_params: [
                 spec: %{String.t() => String.t()},
                 description: """
-                A map with options that will be passed to the encoder.
+                A map with parameters that are passed to the encoder.
+
+                You can use options from: https://ffmpeg.org/ffmpeg-codecs.html#libx264_002c-libx264rgb
+                and https://ffmpeg.org/ffmpeg-codecs.html#Codec-Options
+                Remember not to overwrite options that are available as a separate module options
+                (like `sc_threshold` or `crf`).
```
Comment (mat-hek) [inline lib/membrane_h264_ffmpeg/encoder.ex:134]:
```suggestion
                Options available in the element options (`t:#{inspect(__MODULE__)}.t/0`), like `sc_threshold` or `crf`,
                must be set there and not through `ffmpeg_params`.
```

**Review by mat-hek — APPROVED** [review]

**Pull request merged**